packages/components/src/components/Select/Select.tsx

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+import map from "lodash/map";
+import {
+  Select as SelectPrimitive,
+  SelectArrow,
+  useSelectState,
+  SelectStateProps,
+} from "ariakit/select";
+import type { SelectProps as SelectPrimitiveProps } from "ariakit/select";
+import type { SystemProp, Theme } from "@xstyled/styled-components";
+import { Box } from "../../primitives/Box";
+import { Icon } from "../Icon";
+import { SelectItem } from "./SelectItem";
+import type { SelectItemProps } from "./SelectItem";
+import { SelectPopover } from "./SelectPopover";
+
+export interface SelectProps
+  extends Omit<
+    SelectPrimitiveProps,
+    | "accessibleWhenDisabled"
+    | "autoFocus"
+    | "clickOnEnter"
+    | "clickOnSpace"
+    | "focusable"
+    | "moveOnKeyDown"
+    | "nonce"
+    | "onFocusVisible"
+    | "showOnKeyDown"
+    | "state"
+    | "toggleOnClick"
+    | "toggleOnPress"
+    | "typeahead"
+    | "wrapElement"
+  > {
+  /** The default selected option of the select. */
+  defaultValue?: string;
+  /** Sets the state of the select to disabled, so a user is not able to interact with it. */
+  disabled?: boolean;
+  /** Sets the state of the select to an error state. */
+  hasError?: boolean;
+  /** The `id` of the select. */
+  id?: string;
+  /** The options for the select. */
+  items: SelectItemProps[];
+  /** The `name` of the select. */
+  name?: string;
+  /** Sets the select state to required, so a user has to provide a value in order to be valid. */
+  required?: boolean;
+  /** Function that will be called when setting the select value state. */
+  setValue?: SelectStateProps["setValue"];
+  /** The selected option of the select. */
+  value?: string;
+}
+
+const getSelectStyles = (
+  disabled?: boolean,
+  hasError?: boolean
+): {
+  backgroundColor: SystemProp<keyof Theme["colors"], Theme> | "transparent";
+  borderColor: SystemProp<keyof Theme["colors"], Theme>;
+} => {
+  if (disabled) {
+    return {
+      backgroundColor: "colorBackgroundWeak",
+      borderColor: "colorBorderWeaker",
+    };
+  }
+  if (hasError) {
+    return {
+      backgroundColor: {
+        _: "colorBackground",
+        hover: "colorBackgroundWeakest",
+      },
+      borderColor: "colorBorderError",
+    };
+  }
+  return {
+    backgroundColor: {
+      _: "colorBackground",
+      hover: "colorBackgroundWeakest",
+    },
+    borderColor: "colorBorderWeaker",
+  };
+};
+
+/** A select is an styled dropdown form element that allows users to select a value from a list. */
+const Select = React.forwardRef<HTMLButtonElement, SelectProps>(
```

Comment:
So, we will not have a large size? From what I saw on Figma, we have two variants for size: 

![image](https://user-images.githubusercontent.com/5320963/207295428-ce9ef22f-50f5-4ee0-a1b4-f6d4b64b8b6d.png)

Reply:
Just missed the large size. Adding it now.

Reply:
@apvale large selects are now in storybook. Good catch.